shared/repository/postgresRepository/ticket.go

```diff
@@ -5,62 +5,42 @@ import (
 
 	"github.com/verasthiago/tickets-generator/shared/errors"
 	"github.com/verasthiago/tickets-generator/shared/models"
-	"gorm.io/gorm"
 )
 
+const TICKET_MODEL_NAME = "Ticket"
+
 func (p *PostgresRepository) GetTicketListByUserID(userID string) ([]*models.Ticket, error) {
 	var ticketList []*models.Ticket
-	if record := p.db.Where("owner_id = ?", userID).Find(&ticketList); record.Error != nil {
-		if record.Error == gorm.ErrRecordNotFound {
-			return nil, errors.GenericError{
-				Code:    errors.STATUS_NOT_FOUND,
-				Err:     record.Error,
-				Message: "Ticket not found",
-			}
-		}
-		return nil, record.Error
+	if err := errors.HandleDataNotFoundError(p.db.Where("owner_id = ?", userID).Find(&ticketList).Error, TICKET_MODEL_NAME); err != nil {
```

Comment:
Do u think that error handling should be inside of a db's methods implementation?

Reply:
On that case, it's an error package with all errors related, what we can do is separate by files